packages/node/src/utils/substrate.ts

```diff
@@ -375,12 +377,18 @@ export async function fetchBlocksBatches(
       ? undefined
       : fetchRuntimeVersionRange(api, parentBlockHashs),
   ]);
+
   return blocks.map((block, idx) => {
     const events = blockEvents[idx];
-    const parentSpecVersion =
-      overallSpecVer !== undefined
-        ? overallSpecVer
-        : runtimeVersions[idx].specVersion.toNumber();
+    let parentSpecVersion: number | undefined;
+    if (overallSpecVer !== undefined) {
+      parentSpecVersion = overallSpecVer;
+    } else {
+      parentSpecVersion = runtimeVersions
+        ? runtimeVersions[idx].specVersion.toNumber()
+        : undefined;
+    }
```

Comment:
I think this could be reverted and optional chaining added.

```suggestion
const parentSpecVersion =
      overallSpecVer !== undefined
        ? overallSpecVer
        : runtimeVersions?.[idx].specVersion.toNumber();
```